From 7d3282b269d0a5bea3cd9f91afd87a25a110e129 Mon Sep 17 00:00:00 2001
From: Dave Young <dyoung@redhat.com>
Date: Thu, 22 May 2014 02:42:55 -0400
Subject: [firmware] efi: Fix config_table_type array termination

Message-id: <1400726607-13717-4-git-send-email-dyoung@redhat.com>
Patchwork-id: 80104
O-Subject: [RHEL7.1 PATCH 03/35] x86/efi: Fix config_table_type array termination
Bugzilla: 1080109
RH-Acked-by: Chao WANG <chaowang@redhat.com>
RH-Acked-by: Vivek Goyal <vgoyal@redhat.com>

Resolves: bz1080109

Brew build:
https://brewweb.devel.redhat.com/taskinfo?taskID=7480539

Test: Tested kexec/kdump on HP workstation and qemu.
Later patches depends on this one.

Upstream commit:
commit 722da9d20e7f203ddbb54aba32f95a020d41fc42
Author: Leif Lindholm <leif.lindholm@linaro.org>
Date:   Thu Oct 3 15:42:37 2013 +0100

    x86/efi: Fix config_table_type array termination

    Incorrect use of 0 in terminating entry of arch_tables[] causes the
    following sparse warning,

      arch/x86/platform/efi/efi.c:74:27: sparse: Using plain integer as NULL pointer

    Replace with NULL.

    Signed-off-by: Leif Lindholm <leif.lindholm@linaro.org>
    [ Included sparse warning in commit message. ]
    Signed-off-by: Matt Fleming <matt.fleming@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/platform/efi/efi.c b/arch/x86/platform/efi/efi.c
index bac1ddc..92c0234 100644
--- a/arch/x86/platform/efi/efi.c
+++ b/arch/x86/platform/efi/efi.c
@@ -71,7 +71,7 @@ static __initdata efi_config_table_type_t arch_tables[] = {
 #ifdef CONFIG_X86_UV
  {UV_SYSTEM_TABLE_GUID, "UVsystab", &efi.uv_systab},
 #endif
- {NULL_GUID, NULL, 0},
+ {NULL_GUID, NULL, NULL},
 };
 
 /*
-- 
1.7.1